### ISSUE TITLE
Auth refresh request failed, Unauthorized

### ISSUE THREAD

**Issue by feedmypixel** [backstage/backstage#5942]
**TL;DR** When a logged in user does not browse the app for a while and they then try to browse again they are greeted with the error `401 Error: Auth refresh request failed, Unauthorized`. Our `/api` calls are behind auth so this means a user needs to log in again to browse the application when they haven't browsed it for a while.

## Expected Behavior

When a user:
- Has previously logged in to an Authenticated Backstage application
- Has not browsed the application for a while (longer than 1 hour)
- Has a valid okta refresh token
When a user browses the site they should either:
- Be able to continue browsing the site
- Be asked to log back in again

## Current Behavior

When a user:
- Has previously logged in to an Authenticated Backstage application
- Has not browsed the application for a while (longer than 1 hour)
- Has a valid okta refresh token
- When a user browses the site they get a `401` on the `/api/auth/okta/refresh?optional&env=development` url and the following error in the console:
```
index.esm.js?8d0c:473 Uncaught (in promise) Error: Auth refresh request failed, Unauthorized at DefaultAuthConnector.refreshSession
```
[link to code](https://github.com/backstage/backstage/blob/3aeccc94baf3874a8555b1cdc7c8790b4e63b73b/packages/core-app-api/src/lib/AuthConnector/DefaultAuthConnector.ts#L111)

## Possible Solution

Catch this error and ask users to log in again

## Steps to Reproduce

- Login to app using Okta
- Browse site
- Don't browse site for over an hour
- Browse site again
- Note error

## Context

<img width="878" alt="Screenshot 2021-06-07 at 10 04 09" src="https://user-images.githubusercontent.com/2305016/120994752-b31fa600-c77c-11eb-8101-0b89247cff44.png">
![Screenshot 2021-06-07 at 10 04 26](https://user-images.githubusercontent.com/2305016/120994758-b4e96980-c77c-11eb-9210-8606ee29f4b6.png)
<img width="1270" alt="Screenshot 2021-06-07 at 10 05 13" src="https://user-images.githubusercontent.com/2305016/120994760-b5820000-c77c-11eb-895c-e91ac28c0aba.png">

## Your Environment

We have placed all of our calls too `/api/*` behind auth. 
Anything `/api/auth/*` is not behind auth:
- `/api/auth/okta/*`
- `/api/auth/okta/handler/frame`
- `/api/auth/.well-known/jwks.json`
- `/api/auth/okta/logout`

This is why this is causing us an issue because the bearer is not being supplied to calls to the Catalog when a user doesn't browse the site for a while. Ideally a user should be silently loggedIn again using the refresh token or asked to login again by being presented with the login dialog

**Comment by feedmypixel** [comment]
I suspect this problem may be occurring due to the `.wellknown` changing because of either:
- A new monolith deploy
- A cold starts of the monolith

The user who has not browsed for a while then goes to browse and gets 401s because the `.wellknown` has changed and we are using this `.wellknown` and the `JWT` being sent to `/api/*` calls to auth. 
Not wanting to waste your time and I suspect that something else is going on here other than the code in BackStage, do you have any advice around this? If not I will investigate/dig further this end.

**Comment by freben** [comment]
Pinging @Rugvip in here.

**Comment by feedmypixel** [comment]
I do also see the error:
`Failed to refresh access token [object Object]` and `401` coming back from the refresh url ```refresh?optional&env=development```
<img width="1262" alt="Screenshot 2021-06-09 at 10 12 15" src="https://user-images.githubusercontent.com/2305016/121327612-8057e800-c90b-11eb-9712-8d431d393286.png">

This occurs when I am running my local, and have a valid `Okta` refresh token but haven't browsed for a while. Due to our `/api/` calls being authed any calls to the catalog result in the following: 

![Screenshot 2021-06-09 at 10 13 56](https://user-images.githubusercontent.com/2305016/121327526-6f0edb80-c90b-11eb-9e47-5866640964fe.png)

I will have a look at not using the `.wellknown` from BackStage to auth our `/api` calls today to see if we can solve this problem

**Comment by feedmypixel** [comment]
I also feel that the fundamental issue here is if a request `401`s for whatever reason. The user needs too logout of the app and log back in.

Ideally if there is `401` given it should redirect a user to the login page. I will also have a look at creating this custom work. Do shout if any of this rings any bells.

**Comment by feedmypixel** [comment]
FYI after looking at this in detail today and after the Update I made yesterday up to and including this release https://github.com/backstage/backstage/releases/tag/release-2021-06-07 I am not seeing issues with 401s anymore.

I need to speak with the team in the morning but once thats done I'm pretty sure I will close this ticket

**Comment by freben** [comment]
Nice! Yeah let us know how it goes.

**Comment by feedmypixel** [comment]
Closing this as the problem appears to of been fixed with the latest set of updates - thanks all
